x/farming/types/plan.go

```diff
@@ -209,3 +212,46 @@ type PlanI interface {
 
 	String() string
 }
+
+// ValidatePlanName validates if the plan name exists in store.
+func ValidatePlanName(plans []PlanI, name string) error {
+	for _, plan := range plans {
+		switch p := plan.(type) {
+		case *FixedAmountPlan:
+			if p.Name == name {
+				return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "plan name '%s' already exists", name)
+			}
+		case *RatioPlan:
+			if p.Name == name {
+				return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "plan name '%s' already exists", name)
+			}
+		default:
+			return sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "unrecognized plan type: %T", p)
+		}
+	}
+	return nil
+}
+
+// ValidateOverflowEpochRatio validates that a single account cannot create ratio plans that exceeds
+// a total epoch ratio of 1 (100%) because it will overflow the account's balance
+func ValidateOverflowEpochRatio(plans []PlanI, farmingPoolAddr string, epochRatio sdk.Dec) error {
+	farmerEpochRatio := make(map[string]sdk.Dec)
+
+	for _, plan := range plans {
+		farmingPoolAddr := plan.GetFarmingPoolAddress().String()
+
+		if plan, ok := plan.(*RatioPlan); ok {
+			farmerEpochRatio[farmingPoolAddr] = plan.EpochRatio
```

Comment:
how about add only for validated plans (ex. endTime is not done )

Reply:
I think the map can be overwritten, here also need `if epochRatio, ok := farmerEpochRatio[farmingPoolAddr]; ok {`
for validate the logic, we need some test codes